pkg/auth/api/user/user_actions.go

```diff
@@ -134,6 +138,20 @@ func (h *Handler) setPassword(actionName string, action *types.Action, request *
 		return errors.New("Invalid password")
 	}
 
+	userID := request.Request.Header.Get("Impersonate-User")
```

Comment:
Is this not in the userData ? even the userID ?

Reply:
It is, as the `UserFieldUsername` I'll update.